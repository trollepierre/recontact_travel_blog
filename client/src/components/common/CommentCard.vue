<template>
  <div class="comment-card">
    <article class="comment">
      <div class="comment__content">
        <p>{{ comment.text }}</p>
      </div>
    </article>
  </div>
</template>

<script>
  export default {
    name: 'CommentCard',
    props: {
      comment: { type: Object, default: () => {} },
    },
  }
</script>

<style scoped>
  .comment {
    min-width: 260px;
    background: #ffffff;
    border-radius: 4px !important;
    box-shadow: 0 1px 1px rgba(0, 0, 0, .15);
    border: 1px solid rgba(0, 0, 0, .09);
    display: flex;
    flex-direction: column;
    color: #535a60;
  }

  .comment__content {
    font-size: 15px;
    padding: 15px;
    display: block;
    color: #000;
    text-align: center;
  }

</style>
