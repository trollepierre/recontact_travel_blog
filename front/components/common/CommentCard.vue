<template>
  <div class="comment-card">
    <article class="comment">
      <header class="comment__header">
        <p class="comment__author">
          {{ commentAuthor }}
        </p>
      </header>
      <footer class="comment__footer">
        <p>{{ comment.text }}</p>
      </footer>
    </article>
  </div>
</template>

<script>
  import { formatDateWithLongEndianLongFormat } from '../../services/utils/date/date-utils'
  import translationService from '../../services/services/translations'

  export default {
    name: 'CommentCard',
    props: { comment: { type: Object, default: () => {} } },
    computed: {
      commentAuthor() {
        const date = formatDateWithLongEndianLongFormat(new Date(this.comment.createdAt))
        if (translationService.isFrancophone()) {
          return `De ${this.comment.author} - ${date}`
        }
        return `From ${this.comment.author} - ${date}`
      },
    },
  }
</script>

<style scoped>
  .comment {
    min-width: 260px;
    background: #ffffff;
    border-radius: 4px !important;
    box-shadow: 0 1px 1px rgba(0, 0, 0, .15);
    border: 1px solid rgba(0, 0, 0, .09);
    display: flex;
    flex-direction: column;
    color: #535a60;
  }

  .comment__header {
    border-bottom: 1px solid #e6e6e6;
    padding: 15px;
    height: 34px;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .comment__author {
    font-size: 16px;
    font-weight: 700;
    line-height: 17px;
    color: #07c;
    margin: 0;
    overflow-wrap: break-word;
  }

  .comment__footer {
    text-align: center;
    padding: 15px;
    border-top: 1px solid #e6e6e6;
  }
</style>
