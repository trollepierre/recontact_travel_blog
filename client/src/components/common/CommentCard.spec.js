import CommentCard from './CommentCard.vue'

describe('Component | CommentCard.vue', () => {
  let localVue
  let wrapper
  const propsData = { comment: { text: 'comment', createdAt: '2019-01-13T21:12:37.300Z' } }

  beforeEach(() => {
    localVue = createLocalVue()
  })

  it('should be named "CommentCard"', () => {
    wrapper = shallowMount(CommentCard, { localVue, propsData })

    expect(wrapper.name()).toEqual('CommentCard')
  })

  describe('template', () => {
    it('should match snapshot', () => {
      wrapper = shallowMount(CommentCard, { localVue, propsData })

      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
