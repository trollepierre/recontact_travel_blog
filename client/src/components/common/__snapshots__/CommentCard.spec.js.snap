// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | CommentCard.vue template should match snapshot 1`] = `
<div
  class="comment-card"
>
  <article
    class="comment"
  >
    <div
      class="comment__content"
    >
      <p>
        comment
      </p>
    </div>
  </article>
</div>
`;
